File: src/main/java/com/github/oxisto/JNIStubsCLI.java
package com.github.oxisto;

import com.github.javaparser.JavaParser;
import com.github.javaparser.ParserConfiguration;
import com.github.javaparser.symbolsolver.JavaSymbolSolver;
import com.github.javaparser.symbolsolver.resolution.typesolvers.CombinedTypeSolver;
import com.github.javaparser.symbolsolver.resolution.typesolvers.JavaParserTypeSolver;
import com.github.javaparser.symbolsolver.resolution.typesolvers.ReflectionTypeSolver;
import java.io.File;
import java.io.FileNotFoundException;

public class JNIStubsCLI {

  public static void main(String[] args) {
    // args 0 = root
    // args 1 = class

    var typeResolver = new CombinedTypeSolver();
    ReflectionTypeSolver reflectionTypeSolver = new ReflectionTypeSolver();
    typeResolver.add(reflectionTypeSolver);

    JavaParserTypeSolver javaParserTypeSolver = new JavaParserTypeSolver(new File(args[0]));
    typeResolver.add(javaParserTypeSolver);

    var symbolResolver = new JavaSymbolSolver(typeResolver);
    var parserConfiguration = new ParserConfiguration();
    parserConfiguration.setSymbolResolver(symbolResolver);

    var parser = new JavaParser(parserConfiguration);

    try {
      var cu = parser.parse(new File(args[0] + "/" + args[1]));
      var o = cu.getResult();
      if (o.isPresent()) {
        var result = o.get();

        for (var type : result.getTypes()) {
          var header = new HeaderGenerator(type);
          System.out.println(header.generate());

          var source = new SourceGenerator(type);
          System.out.println(source.generate());
        }
      }

    } catch (FileNotFoundException e) {
      e.printStackTrace();
    }
  }
}

File: src/main/java/com/github/oxisto/jnistubs/JNIStubsCLI.kt
package com.github.oxisto.jnistubs

import com.github.javaparser.JavaParser
import com.github.javaparser.ParserConfiguration
import com.github.javaparser.symbolsolver.JavaSymbolSolver
import com.github.javaparser.symbolsolver.resolution.typesolvers.CombinedTypeSolver
import com.github.javaparser.symbolsolver.resolution.typesolvers.JavaParserTypeSolver
import com.github.javaparser.symbolsolver.resolution.typesolvers.ReflectionTypeSolver
import java.io.File
import java.io.FileNotFoundException

object JNIStubsCLI {

  @JvmStatic
  fun main(args: Array<String>) {
    // args 0 = root
    // args 1 = class

    val typeResolver = CombinedTypeSolver()
    val reflectionTypeSolver = ReflectionTypeSolver()
    typeResolver.add(reflectionTypeSolver)

    val javaParserTypeSolver = JavaParserTypeSolver(File(args[0]))
    typeResolver.add(javaParserTypeSolver)

    val symbolResolver = JavaSymbolSolver(typeResolver)
    val parserConfiguration = ParserConfiguration()
    parserConfiguration.setSymbolResolver(symbolResolver)

    val parser = JavaParser(parserConfiguration)

    try {
      val cu = parser.parse(File(args[0] + "/" + args[1]))
      val o = cu.result
      if (o.isPresent) {
        val result = o.get()

        for (type in result.types) {
          val header = HeaderGenerator(type)
          println(header.generate())

          val source = SourceGenerator(type)
          println(source.generate())
        }
      }

    } catch (e: FileNotFoundException) {
      e.printStackTrace()
    }

  }
}
